cabal-install/Distribution/Client/Setup.hs

```diff
@@ -2389,7 +2389,7 @@ data SDistExFlags = SDistExFlags {
   }
   deriving (Show, Generic)
 
-data ArchiveFormat = TargzFormat | ZipFormat -- ...
+data ArchiveFormat = TargzFormat -- ...
```

Comment:
At this point, we might as well drop the `data ArchiveFormat` altogether.

@23Skidoo what do you think? is there any reason to keep this datatype now that there's no choice anymore?